src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/SourceFileHandlingTests.cs

```diff
@@ -55,6 +193,284 @@ public void AddRemoveAdditionalFile_CPS()
             Assert.Empty(GetCurrentAdditionalDocuments());
         }
 
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
+        public void RenameAdditionalFile_CPS()
+        {
+            using var environment = new TestEnvironment();
+            using var project = CreateCSharpCPSProject(environment, "project1");
+            TextDocument GetSingleDocument() => environment.Workspace.CurrentSolution.Projects.Single().Documents.SingleOrDefault();
+
+            Assert.Null(GetSingleDocument());
```

Comment:
Do these asserts really help anything? Otherwise you can just change the call to .Single() and then remove all the not-null asserts everywhere.

Reply:
Personal taste, but asserting the state before and after change is not bad. Not needed for state of everything, but for the state that will change I think it's worth. 

i.e: I'm proving there was no document before other modifications